File: 0429-n-ary-tree-level-order-traversal/recursive-depth-first-search/python/solution.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        res = []

        def dfs(node, level):
            if not node:
                return
            if len(res) == level:
                res.append([])
            res[level].append(node.val)
            for child in node.children:
                dfs(child, level + 1)

        dfs(root, 0)
        return res

File: 0429-n-ary-tree-level-order-traversal/recursive-depth-first-search/cpp/solution.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> res;
        dfs(root, 0, res);
        return res;
    }

private:
    void dfs(Node* node, int level, vector<vector<int>>& res) {
        if (!node) {
            return;
        }

        if (res.size() == level) {
            res.push_back({});
        }
        
        res[level].push_back(node->val);
        for (Node* child : node->children) {
            dfs(child, level + 1, res);
        }
    }
};
